### PR TITLE
MES-4427: Creating test report for a mod2

### PR DESCRIPTION
**Pull request by szabikr**
## Description

Creating test report for category AMod2

## Checklist

- [ ] PR title includes the JIRA ticket number
- [ ] Branch is rebased against the latest develop
- [ ] Code has been tested manually
- [ ] PR link added to JIRA ticket
- [ ] One review from each scrum team
- [ ] Squashed commit contains the JIRA ticket number

### CODE REVIEW

File: src/pages/test-report/cat-a-mod2/components/vehicle-checks/__tests__/vehicle-checks.spec.ts [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+  }));
+
+  describe('Class', () => {
+    const vehicleChecksScore: VehicleChecksScore = {
+      drivingFaults: 4,
+      seriousFaults: 1,
+    };
+
+    beforeEach(() => {
+      spyOn(component.faultCountProvider, 'getVehicleChecksFaultCountCatBE').and.returnValue(vehicleChecksScore);
```
Comment (MarkPatt89) [inline src/pages/test-report/cat-a-mod2/components/vehicle-checks/__tests__/vehicle-checks.spec.ts:60]: todo need for the spy.

File: src/pages/test-report/cat-a-mod2/components/vehicle-checks/__tests__/vehicle-checks.spec.ts [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+    });
+
+    it('should set the vehicle checks driving fault count', (done: DoneFn) => {
+      component.ngOnInit();
+      component.componentState.vehicleChecksDrivingFaultCount$.subscribe((result) => {
+        expect(component.faultCountProvider.getVehicleChecksFaultCountCatBE).toHaveBeenCalled();
```
Comment (MarkPatt89) [inline src/pages/test-report/cat-a-mod2/components/vehicle-checks/__tests__/vehicle-checks.spec.ts:67]: todo needed for getVehicleChecksFaultCount throughout this spec file.

File: src/pages/test-report/cat-a-mod2/components/vehicle-checks/vehicle-checks.ts [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+        select(getVehicleChecksCatBE),
+        map((vehicleChecks) => {
+
+          // TODO - PREP-AMOD2: Change provider function to a cat a mod2 one
+          return this.faultCountProvider.getVehicleChecksFaultCountCatBE(vehicleChecks).drivingFaults;
+        }),
+      ),
+      vehicleChecksSeriousFaultCount$: currentTest$.pipe(
+        select(getTestData),
+        select(getVehicleChecksCatBE),
```
Comment (MarkPatt89) [inline src/pages/test-report/cat-a-mod2/components/vehicle-checks/vehicle-checks.ts:53]: todo needed for getVehicleChecksCatAMod2.

File: src/pages/test-report/cat-a-mod2/test-report.cat-a-mod2.page.ts [hunk 4]
```diff
@@ -0,0 +1,249 @@
...
+} from '../../../modules/tests/test-data/cat-be/test-requirements/test-requirements.cat-be.reducer';
+import { legalRequirementsLabels } from '../../../shared/constants/legal-requirements/legal-requirements.constants';
+
+interface TestReportPageState {
+  candidateUntitledName$: Observable<string>;
+  isRemoveFaultMode$: Observable<boolean>;
+  isSeriousMode$: Observable<boolean>;
+  isDangerousMode$: Observable<boolean>;
+  manoeuvres$: Observable<boolean>;
+  testData$: Observable<CatBEUniqueTypes.TestData>;
```
Comment (MarkPatt89) [inline src/pages/test-report/cat-a-mod2/test-report.cat-a-mod2.page.ts:68]: todo needed for testData$

Comment (MarkPatt89) [reply src/pages/test-report/cat-a-mod2/test-report.cat-a-mod2.page.ts:68]: and also for testRequirements$

**Review by MarkPatt89 — COMMENTED** [review]
A few todo's outstanding, happy to approve when they're added.

File: src/pages/test-report/cat-a-mod2/components/vehicle-checks/__tests__/vehicle-checks.spec.ts (hunk 1 above)
Comment (szabikr) [reply src/pages/test-report/cat-a-mod2/components/vehicle-checks/__tests__/vehicle-checks.spec.ts:60]: added

File: src/pages/test-report/cat-a-mod2/components/vehicle-checks/vehicle-checks.ts (hunk 3 above)
Comment (szabikr) [reply src/pages/test-report/cat-a-mod2/components/vehicle-checks/vehicle-checks.ts:53]: added

File: src/pages/test-report/cat-a-mod2/test-report.cat-a-mod2.page.ts (hunk 4 above)
Comment (szabikr) [reply src/pages/test-report/cat-a-mod2/test-report.cat-a-mod2.page.ts:68]: added them

File: src/pages/test-report/cat-a-mod2/components/vehicle-checks/__tests__/vehicle-checks.spec.ts (hunk 2 above)
Comment (szabikr) [reply src/pages/test-report/cat-a-mod2/components/vehicle-checks/__tests__/vehicle-checks.spec.ts:67]: added

**Review by amobarak — APPROVED** [review]

**Review by MarkPatt89 — APPROVED** [review]

**Pull request merged**